### ISSUE TITLE
Weaviate index ignores supplied prompt on query

### ISSUE THREAD

**Issue by cwelton** [jerryjliu/gpt_index#365]
While it is documented as if you can supply a prompt to a weaviate index the supplied prompt is NOT used on query.

Instead a QueryRunner instance is created based on 
query_runner.py line 101:
```
        query_cls = self._index_registry.type_to_query[index_struct_type][mode]
        # if recursive, pass self as query_runner to each individual query
        query_runner = self if self._recursive else None
        query_kwargs = self._get_query_kwargs(config)
        query_obj = query_cls(
            index_struct,
            **query_kwargs,
            query_runner=query_runner,
            docstore=self._docstore,
        )
```
Where query_cls is a base `GPTWeaviateIndexQuery` and it gets initialized without passing down the parent index's query prompt, seemingly with no way of passing the prompt down

Repo:
```
prompt_tmpl = """
Context information is below.
---------------------
{context_str}
---------------------
Given the above context answer the following question.  Only answer "Yes", "No", or "That is not a yes or no question".

Question: {query_str}
"""
prompt =  QuestionAnswerPrompt(prompt_tmpl)
windex =  GPTWeaviateIndex(documents, weaviate_client=wclient, text_qa_prompt=prompt_tmpl)
resp = windex.query('what is the meaning of life?')
print(resp.response)
```

Unsuccessful attempted workaround:
```
resp = windex.query('what is the meaning of life?', text_qa_template=windex.text_qa_template)
```

**Comment by jerryjliu** [comment]
@cwelton thanks for surfacing this, apologies for the delayed response. This sounds like a bug - i'll put out a fix

**Comment by jerryjliu** [comment]
should be fixed in #639 - if you specify a custom prompt during index construction, that will apply during query-time (the new proper flow is to specify these prompts during query-time)
